repose-aggregator/commons/utilities/src/main/java/org/openrepose/commons/utils/logging/apache/format/stock/TimeReceivedHandler.java

```diff
@@ -28,8 +29,20 @@
 
 public class TimeReceivedHandler implements FormatterLogic {
 
+    private final static String DEFAULT_DATE_FORMAT = "dd-MM-yyyy-HH:mm:ss.SSS";
+
+    private final String dateFormat;
+
+    public TimeReceivedHandler(final String dateFormat) {
+        this.dateFormat = StringUtils.isEmpty(dateFormat) ? DEFAULT_DATE_FORMAT : dateFormat;
+    }
+
     @Override
     public String handle(HttpServletRequest request, HttpServletResponse response) {
-        return new SimpleDateFormat("dd-MM-yyyy-HH:mm:ss.SSS").format(Calendar.getInstance().getTime());
+        return new SimpleDateFormat(dateFormat).format(Calendar.getInstance().getTime());
```

Comment:
the constructor errors out if it's a bad format right? if so i'd suggest building it in the constructor, that failure should chain up and prevent repose from starting, instead of bombing after the request has been successfully processed by the origin service.